### PR TITLE
COM-1065 Add brevo configuration page

### PR DESCRIPTION
**Pull request by juliawegmayr**
## Description

Add a config page to enable admins to configure their brevo settings instead of handling it in the code.
In this PR only the senders are handled. Additional fields will be added in upcoming PRs.

## Screenshots/screencasts

Brevo config page:
Screenshot 2024-10-17 at 14 09 12

Send Manager, if config is not set:
Screenshot 2024-10-17 at 14 11 01

## Changeset

[x] I have verified if my change requires a changeset

## Related tasks and documents

https://vivid-planet.atlassian.net/browse/COM-1065

## Further information

There was already a merge request for adding a brevo configuration page, which was removed afterwards. After deciding, that configuring should be done in comet, we need to implement the configuration page again: https://github.com/vivid-planet/comet-brevo-module/pull/45

### CODE REVIEW

File: demo/api/src/db/migrations/Migration20241016123307.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import { Migration } from "@mikro-orm/migrations";
+
+export class Migration20241016123307 extends Migration {
+    async up(): Promise<void> {
+        this.addSql(
+            'create table "BrevoConfig" ("id" uuid not null, "createdAt" timestamp with time zone not null, "updatedAt" timestamp with time zone not null, "scope_domain" text not null, "scope_language" text not null, "senderMail" text not null, "senderName" text not null, constraint "BrevoConfig_pkey" primary key ("id"));',
+        );
+    }
+}
```
Comment (RainbowBunchie) [inline demo/api/src/db/migrations/Migration20241016123307.ts:9]: This migration should be included in the module's migrations

File: packages/api/src/brevo-config/brevo-config.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        @Query(() => BrevoConfig, { nullable: true })
+        async brevoConfig(
+            @Args("scope", { type: () => Scope }, new DynamicDtoValidationPipe(Scope))
+            scope: typeof Scope,
+        ): Promise<BrevoConfigInterface | null> {
+            const brevoConfig = await this.repository.findOne({ scope });
+            return brevoConfig;
+        }
+
+        // TODO: add validation if the input contains a valid sender
```
Comment (RainbowBunchie) [inline packages/api/src/brevo-config/brevo-config.resolver.ts:56]: Should this be done in this MR or will there be a follow up?

File: packages/api/src/brevo-config/brevo-config.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            const senders = await this.brevoSenderApiService.getSenders(Scope);
+
+            if (senders && senders.some((sender) => sender.email === email && sender.name === name)) {
+                return true;
+            }
+
+            return false;
+        }
+
+        @RequiredPermission(["brevo-newsletter-config"], { skipScopeCheck: true })
```
Comment (RainbowBunchie) [inline packages/api/src/brevo-config/brevo-config.resolver.ts:40]: Why is the skipScope check necessary here?

File: packages/api/src/brevo-config/brevo-config.resolver.ts (hunk 2 above)
Comment (RainbowBunchie) [reply packages/api/src/brevo-config/brevo-config.resolver.ts:56]: you already defined `isValidSender`

**Review by RainbowBunchie — CHANGES_REQUESTED** [review]

File: demo/api/src/db/migrations/Migration20241016123307.ts (hunk 1 above)
Comment (juliawegmayr) [reply demo/api/src/db/migrations/Migration20241016123307.ts]: Corrected here: 2540ea3d97fda5e54b154c93cd847858b05eea85

File: packages/api/src/brevo-config/brevo-config.resolver.ts (hunk 2 above)
Comment (juliawegmayr) [reply packages/api/src/brevo-config/brevo-config.resolver.ts]: Added here: cc8900cc784b81ede87a2899dd4ca671cf82493a

File: packages/api/src/brevo-config/brevo-config.resolver.ts (hunk 3 above)
Comment (juliawegmayr) [reply packages/api/src/brevo-config/brevo-config.resolver.ts]: Corrected: c01b7a05fb5f2cf03f09c88654f6838281b0f472

File: packages/api/src/brevo-config/brevo-config.resolver.ts (hunk 2 above)
Comment (RainbowBunchie) [reply packages/api/src/brevo-config/brevo-config.resolver.ts]: This should also be checked on update

Comment (juliawegmayr) [reply packages/api/src/brevo-config/brevo-config.resolver.ts]: Of course, sorry! Added it here: afc700525b410f25a02439378d0fb6399f565118

**Comment by RainbowBunchie** [comment]
LGTM please resolve conflicts and @raphaelblum please review 🙏🏻

**Comment by juliawegmayr** [comment]
@RainbowBunchie @raphaelblum branch is updated to current next branch (comet 7) and ready from my side :)

**Review by raphaelblum — APPROVED** [review]

File: demo/campaign/next-env.d.ts [hunk 4]
```diff
@@ -2,4 +2,4 @@
 /// <reference types="next/image-types/global" />
 
 // NOTE: This file should not be edited
-// see https://nextjs.org/docs/basic-features/typescript for more information.
+// see https://nextjs.org/docs/pages/building-your-application/configuring/typescript for more information.
```
[the remark this answers is not in the document]
Comment (RainbowBunchie) [reply demo/campaign/next-env.d.ts:5]: Either this was forgotten in the v7 update or this should not be changes

**Review by RainbowBunchie — APPROVED** [review]

**Pull request merged**